torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -355,8 +386,14 @@ void PropagateShapeOnNode(Node * node, bool insert_expands) {
     default:
       break; // fall-through
   }
-  if (node->matches("aten::cat(Tensor[] tensors, int dim) -> Tensor")) {
-    return PropagateCatShape(node);
+
+  if (node->matches("aten::cat(Tensor[] tensors, int dim) -> Tensor")){
+      return PropagateCatShape(node, false);
+  }
+  if (node->matches("aten::cat(Tensor[] tensors, int dim, Scalar pad_value) -> Tensor")) {
+    at::Scalar pad_value = *node->get<at::Scalar>(attr::pad_value);
```

Comment:
I'm getting the following error when building after merging with latest master:

```
../torch/csrc/jit/autodiff.cpp:333:61: error: ‘pad_value’ is not a member of ‘torch::jit::attr’
         at::Scalar pad_value = *node->get<at::Scalar>(attr::pad_value);
                                                             ^~~~~~~~~
../torch/csrc/jit/autodiff.cpp:333:61: note: suggested alternative: ‘max_values’
         at::Scalar pad_value = *node->get<at::Scalar>(attr::pad_value);
                                                             ^~~~~~~~~
                                                             max_values
```
I think you need to add an the attr to `aten/src/ATen/core/interned_strings.h`

Reply:
It is declared in torch/csrc/jit/generated/aten_interned_strings.h. You may need to rebuild to auto-generated that file.